app/src/lib/stores/accounts-store.ts

```diff
@@ -83,7 +83,27 @@ export class AccountsStore extends BaseStore {
       )
     } catch (e) {
       log.error(`Error adding account '${account.login}'`, e)
-      this.emitError(e)
+
+      if (__DARWIN__) {
+        const error = e as Error
+
+        if (
+          error.message &&
+          error.message.startsWith(
+            'The user name or passphrase you entered is not correct'
+          )
```

Comment:
Could we extract this check into a helper function? `IsKeychainError` or something? That way we should be able to clean this method up and get rid of a bunch of these nested statements to something similar to

```ts
catch (e) {
  if (__DARWIN__ && isKeyChainError(e)) {
    this.emitError(new Error(...))
  } else {
     this.emitError(...)
  }
}
```